sdk/src/main/scala/com/horizen/api/http/SidechainCswApiRoute.scala

```diff
@@ -0,0 +1,163 @@
+          }
+        case Failure(e) =>
+          log.error("Unexpected error during CSW proof generation.")
+          ApiResponseUtil.toResponse(ErrorCswGenerationState("Unexpected error during CSW proof generation.", JOptional.of(e)))
+      }
+    }
+  }
+
+  /**
+   * return the csw info for given BoxId
+   */
+  def cswInfo: Route = (post & path("cswInfo")) {
+     entity(as[ReqCswInfo]) { body =>
+       Try {
+         Await.result(cswManager ? GetCswInfo(BytesUtils.fromHexString(body.boxId)), timeout.duration).asInstanceOf[CswInfo]
+       } match {
+         case Success(cswInfo: CswInfo) =>
+           ApiResponseUtil.toResponse(RespCswInfo(cswInfo))
+         case Failure(e) =>
+           log.error("Unexpected error during retrieving CSW info.")
+           ApiResponseUtil.toResponse(ErrorRetrievingCeasingState("Unexpected error during retrieving CSW info.", JOptional.of(e)))
+       }
+     }
+  }
+
+  /**
+   * Return the list with all box ids.
+   */
+  def cswBoxIds: Route = (post & path("cswBoxIds")) {
+    Try {
+      Await.result(cswManager ? GetCswBoxIds, timeout.duration).asInstanceOf[Seq[Array[Byte]]]
+    } match {
+      case Success(boxIds: Seq[Array[Byte]]) => {
+        val boxIdsStr = boxIds.map(id => BytesUtils.toHexString(id))
+        ApiResponseUtil.toResponse(RespCswBoxIds(boxIdsStr))
+      }
+      case Failure(e) =>
+        log.error("Unexpected error during retrieving CSW Box Ids.")
+        ApiResponseUtil.toResponse(ErrorRetrievingCeasingState("Unexpected error during retrieving CSW Box Ids.", JOptional.of(e)))
+    }
+  }
+
+  /**
+   * Return the nullifier for the given coin box id.
+   */
+  def nullifier: Route = (post & path("nullifier")) {
+    entity(as[ReqNullifier]) { body =>
+      Try {
+        Await.result(cswManager ? GetBoxNullifier(BytesUtils.fromHexString(body.boxId)), timeout.duration).asInstanceOf[Array[Byte]]
+      } match {
+        case Success(nullifier: Array[Byte]) =>
+          ApiResponseUtil.toResponse(RespNullifier(BytesUtils.toHexString(nullifier)))
+        case Failure(e) =>
+          log.error("Unexpected error during retrieving the nullifier.")
+          ApiResponseUtil.toResponse(ErrorRetrievingCeasingState("Unexpected error during retrieving the nullifier.", JOptional.of(e)))
+      }
+    }
+  }
+}
+
+object SidechainCswRestScheme {
+  @JsonView(Array(classOf[Views.Default]))
+  private[api] case class RespCswHasCeasedState(state: Boolean) extends SuccessResponse
+
+  @JsonView(Array(classOf[Views.Default]))
+  private[api] case class ReqGenerationCswState(boxId: String, mcAddress: String) {
+    require(boxId.length == 64, s"Invalid id $boxId. Id length must be 64")
+    require(mcAddress.length == 64, s"Invalid address $mcAddress. Address length must be 64")
+  }
+
+  @JsonView(Array(classOf[Views.Default]))
+  private[api] case class RespGenerationCswState(state: String) extends SuccessResponse
```

Comment:
Let's change it with `state: String, description: String`.
Current answers like "Sidechain is alive" is a `description`, "SidechainIsAlive" is a `state`.
It will allow both users to read and programs to parse the state.